server/routes/disasters.py

```diff
@@ -67,9 +68,25 @@ def disaster_dashboard(place_dcid=DEFAULT_PLACE_DCID):
   place_type = DEFAULT_PLACE_TYPE
   if place_dcid != DEFAULT_PLACE_DCID:
     place_type = place_api.get_place_type(place_dcid)
+    if not place_type:
+      place_type = "Place"
   place_name = place_api.get_i18n_name([place_dcid
                                        ]).get(place_dcid, escape(place_dcid))
 
+  all_stat_vars = lib_page_config.get_all_variables(dashboard_config)
+  stat_vars_existence = dc.observation_existence(all_stat_vars, [place_dcid])
+
+  for stat_var in stat_vars_existence['variable']:
+    if not stat_vars_existence['variable'][stat_var]['entity'][place_dcid]:
+      # This is for the main place, only remove the tile type for single place.
+      for tile_type in [
+          subject_page_pb2.Tile.TileType.HISTOGRAM,
+          subject_page_pb2.Tile.TileType.LINE,
+          subject_page_pb2.Tile.TileType.BAR,
```

Comment:
could you add MAP here too?

Reply:
Here data existence checks for the main place. The map tile would need to check for child places, which I plan to cover with StatVarSummary cache, by matching the parent and child place type (as an approximate)